File: LeetCode/Simulation/Python/Leetcode388-文件的最长绝对路径.py
class Solution:
    def lengthLongestPath(self, input: str) -> int:
        res = 0
        i,tot,stk,res = 0,0,[],0
        while i < len(input):
            k = 0
            while i < len(input) and input[i] == '\t':
                i += 1
                k += 1
            while k < len(stk):
                tot -= stk.pop()
            j = i
            while j < len(input) and input[j] != '\n':
                j += 1
            length = j - i
            stk.append(length)
            tot += length      
            if '.' in input[i:j]:
                res = max(res,tot + len(stk) - 1)
            i = j + 1
        return res

File: LeetCode/Simulation/JS/Leetcode388-文件的最长绝对路径.js
var lengthLongestPath = function (input) {
    const stk = [];
    let res = 0;
    for (let i = 0, sum = 0; i < input.length; i++) {
        let k = 0;
        while (i < input.length && input[i] == '\t') i++, k++;
        while (k < stk.length) sum -= stk.pop();
        let j = i;
        while (j < input.length && input[j] != '\n') j++;
        let len = j - i;
        stk.push(len), sum += len;
        if (input.substr(i, len).indexOf('.') != -1) {
            res = Math.max(res, sum + stk.length - 1);
        }
        i = j;
    }
    return res;
};